src/map_extras.cpp

```diff
@@ -76,6 +76,164 @@ void mx_null( map &, const tripoint & )
     debugmsg( "Tried to generate null map extra." );
 }
 
+void mx_roa_crash( map &m, const tripoint &abs_sub )
+{
+    int cx = rng( 6, SEEX * 2 - 7 );
+    int cy = rng( 6, SEEY * 2 - 7 );
+
+    for( int x = 0; x < SEEX * 2; x++ ) {
+        for( int y = 0; y < SEEY * 2; y++ ) {
+            if( m.veh_at( tripoint( x,  y, abs_sub.z ) ) &&
+                m.ter( tripoint( x, y, abs_sub.z ) )->has_flag( TFLAG_DIGGABLE ) ) {
+                m.ter_set( tripoint( x, y, abs_sub.z ), t_dirtmound );
+            } else {
+                if( x >= cx - dice( 1, 5 ) && x <= cx + dice( 1, 5 ) && y >= cy - dice( 1, 5 ) &&
+                    y <= cy + dice( 1, 5 ) ) {
+                    if( one_in( 7 ) && m.ter( tripoint( x, y, abs_sub.z ) )->has_flag( TFLAG_DIGGABLE ) ) {
+                        m.ter_set( tripoint( x, y, abs_sub.z ), t_dirtmound );
+                    }
+                }
+                if( x >= cx - dice( 1, 6 ) && x <= cx + dice( 1, 6 ) && y >= cy - dice( 1, 6 ) &&
+                    y <= cy + dice( 1, 6 ) ) {
+                    if( !one_in( 5 ) ) {
+                        m.make_rubble( tripoint( x,  y, abs_sub.z ), f_wreckage, true );
+                        if( m.ter( tripoint( x, y, abs_sub.z ) )->has_flag( TFLAG_DIGGABLE ) ) {
```

Comment:
And there is actually a function to check for a terrain flag:
```suggestion
                        if( m.has_flag_ter( TFLAG_DIGGABLE, tripoint( x, y, abs_sub.z ) ) ) {
```